cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -511,4 +511,13 @@ class ndarray : public ndview<T, axis_count, order> {
     shared_t data_;
 };
 
+#ifdef ONEDAL_DATA_PARALLEL
+template <typename Float, template <typename> typename Accessor, typename Table>
+inline ndarray<Float, 2> flatten_table(sycl::queue& q, const Table& table, sycl::usm::alloc alloc) {
+    Accessor<const Float> accessor{ table };
+    const auto data = accessor.pull(q, { 0, -1 }, alloc);
```

Comment:
This implementation works only for `row_accessor`. To make this work with other accessors, we need to pass indexes from the user to this function directly. Otherwise, `Accessor` template parameter is not needed.